src/main/java/org/myrobotlab/service/JMonkeyEngine.java

```diff
@@ -1468,14 +1468,18 @@ public void onPointCloud(PointCloud pc) {
   }
 
   // auto Register
-  public void onRegistered(Registration registration) throws Exception {
-    // new service - see if we can virtualize it
-    log.info("{}.onRegistered({})", getName(), registration);
-    if (registration.getName().contentEquals("i01.head.jaw")) {
-      log.info("here");
-    }
-    if (autoAttach) {
-      attach(registration.getFullName());
+  public void onRegistered(Registration registration) {

```

Comment:
is this a notification of all services that are registered?  Does this get invoked for every service that gets created?  Does that mean that the number of times that onRegistered gets called its O^2 on the number of services?  ouch.

Reply:
Just O(n).